frontend/src/features/admin-form/responses/FeedbackPage/FeedbackPage.tsx

```diff
@@ -102,3 +228,135 @@ export const FeedbackPage = (): JSX.Element => {
     </Container>
   )
 }
+
+const getReviewInformationComponent = (
+  average: string | undefined,
+  count: number | undefined,
+): JSX.Element => {
+  return (
+    <Grid
+      gridTemplateColumns={{ base: 'auto', md: 'auto auto' }}
+      gridGap={{ base: '0.5rem', md: '1.5rem' }}
+      gridTemplateAreas={{
+        base: "'submissions' 'score'",
+        md: "'score submissions'",
+      }}
+    >
+      <Flex gridArea="score" flexDir="column">
+        <Text textStyle="caption-2" color="secondary.400">
+          Average Score
+        </Text>
+        <Text textStyle="display-2">
+          {average ? Number(average).toPrecision(2) : '-.--'}
+        </Text>
+      </Flex>
+      <Box gridArea="submissions" alignSelf="end">
+        <Text textStyle="h4" mb="0.5rem">
+          <Text as="span" color="primary.500">
+            {count}
+          </Text>
+          {simplur` ${[count || 0]}feedback submission[|s] to date`}
+        </Text>
+      </Box>
+    </Grid>
+  )
+}
+
+const getIssueInformationComponent = (
+  count: number | undefined,
+): JSX.Element => {
+  return (
+    <div>
+      <Box>
+        <Text textStyle="h4" mb="0.5rem">
+          <Text as="span" color="primary.500">
+            {count}
+          </Text>
+          {simplur` ${[count || 0]}issue[|s] to date`}
+          <Tooltip
+            label={`Feedback displayed here relates to form submission issues`}
+            placement="top"
+            textAlign="center"
+          >
+            <Icon as={BxsInfoCircle} aria-hidden marginX="0.5rem" />
+          </Tooltip>
+        </Text>
+      </Box>
+    </div>
+  )
+}
+
+const getInformationGridComponent = (
+  currentFeedbackType: FeedbackType,
+  issueProps: Issue,
+  reviewProps: Review,
+): JSX.Element => {
+  if (currentFeedbackType === FeedbackType.Issues) {
+    return getIssueInformationComponent(issueProps.count)
+  }
+  return getReviewInformationComponent(reviewProps.average, reviewProps.count)
+}
+
+const getFeedbackTypeButtonProps = (
+  currentFeedbackType: FeedbackType,
+  requiredFeedbackType: FeedbackType,
+): ButtonProps => {
+  if (currentFeedbackType === requiredFeedbackType) {
+    return {
+      colorScheme: 'primary',
+      isActive: true,
+    }
+  }
+  return {
+    colorScheme: 'secondary',
+    isActive: false,
+  }
+}
+
+const getFeedBackDownloadButtonProps = (
+  currentFeedbackType: FeedbackType,
+  issueProps: Issue,
+  reviewProps: Review,
+): {
```

Comment:
Maybe we can refactor this to pass only 2 args into it - 1) the respective isLoading state and 2) the count. The logic can be abstracted to be outside of this function itself after all.

Reply:
My thought is that the caller is already very cluttered, I want to abstract the nitty gritty details out, by looking at the original caller, we do not need to know how the props are constructed, just that it is constructed. lmk wdyt

Reply:
Okay I see what you mean. Since there's a lot of repeated if/else logic for for `currentFeedbackType`, how about we use a utility function across all these? Something like this:
```ts
const getFeedbackProps = (
  currentFeedbackType: FeedbackType,
  issueProps: Issue,
  reviewProps: Review,
): {
  if (currentFeedbackType === FeedbackType.Issues) return issueProps
  else return reviewProps
}
```
Then we can use this across all the other functions that are dependent on the `currentFeedbackType` haha

Reply:
we could do that, but i am not sure what is the added value it brings here?

Reply:
To simplify the code since its repeated conditionals that we're doing. I think its cleaner this way at least, since we only do the if/else here in this 1 functions instead of repeating it across all the different ones. Then the existing functions can be simplified to perform exactly what they're supposed to do without the conditional.
Just imo, wdyt?

Reply:
but there is no inheritance to both props, so the caller will still need to check the type of the props to tell which property to use?